common/sgx/eeid_verifier.c

```diff
@@ -321,6 +332,22 @@ static oe_result_t _eeid_verify_evidence(
     /* EEID passed to the verifier */
     if (endorsements_buffer)
     {
+        // Verify and disard attestation header
+        oe_attestation_header_t* header =
+            (oe_attestation_header_t*)endorsements_buffer;
+
+        OE_CHECK(oe_verify_attestation_header(
+            endorsements_buffer, endorsements_buffer_size));
+
+        if (memcmp(
+                &context->base.format_id,
+                &header->format_id,
+                sizeof(oe_uuid_t)))
+            OE_RAISE(OE_CONSTRAINT_FAILED);
+
+        endorsements_buffer += sizeof(*header);
```

Comment:
In attest_plugin.c, endorsements_buffer is not treated as format of         oe_attestation_header_t. Why is it treated as such here?

Reply:
For EEID, the attestation header was wrapped around the endorsements data in enclave/sgx/eeid_attester.c

Reply:
Just a question: why does it need to be wrapped at all?

Reply:
eeid_attester.c wraps endorsements in its own structure oe_eeid_t. The original code seems fine.